src/report.ts

```diff
@@ -4,9 +4,17 @@ import * as models from 'powerbi-models';
 import * as wpmp from 'window-post-message-proxy';
 import * as hpm from 'http-post-message';
 import * as utils from './util';
+import { IFilterable } from './ifilterable';
+import { IPage, Page } from './page';
 
-export class Report extends embed.Embed {
-    static allowedEvents = ["dataSelected", "filterAdded", "filterUpdated", "filterRemoved", "pageChanged", "error"];
+export interface IReport {
+    iframe: HTMLIFrameElement;
+    service: service.IService;
+    config: embed.IInternalEmbedConfiguration
+}
+
+export class Report extends embed.Embed implements IReport, IFilterable {
```

Comment:
Hmm, I was originally looking at the interfaces as not representations of the classes, but the minimal part of the class that the dependencies need.

For instance, in order for the Page class to send requests, it only needs to know about the iframe (for the contentWindow), the service (for hpm), and config (for uniqueId).

If we added IFilterable to these interfaces it means these extra methods would need to be stubbed even though they should never be called.  I'm not sure what is correct to do.  It seems there is conflict of interest between keeping the interfaces with minimal needs for consumers  VS following the convention that IReport should represent the operations that can be done on Report class.

Does this other persepctive sway your opinion or would your instincts at looking at this still make you want IFilterable on the embed object interfaces?